app/src/main/java/org/dhis2/usescases/main/program/HomeRepositoryImpl.kt

```diff
@@ -57,60 +57,21 @@ internal class HomeRepositoryImpl(private val d2: D2, private val eventLabel: St
-                            d2
-                                .categoryModule()
-                                .categoryCombos()
-                                .withCategoryOptionCombos()
-                                .uid(it.categoryCombo()!!.uid())
-                                .blockingGet()!!
-                                .categoryOptionCombos()!!
-                        )
-                    } else {
-                        UidsHelper.getUidsList<CategoryOptionCombo>(
-                            d2.categoryModule().categoryCombos().withCategoryOptionCombos().uid(
-                                dataElement!!.categoryComboUid()
-                            ).blockingGet()!!.categoryOptionCombos()!!
-                        )
-                    }
-                    val attributeOptionCombos = UidsHelper.getUidsList<CategoryOptionCombo>(
-                        d2
-                            .categoryModule()
-                            .categoryCombos()
-                            .withCategoryOptionCombos()
-                            .uid(dataSet.categoryCombo()!!.uid())
-                            .blockingGet()!!
-                            .categoryOptionCombos()!!
-                    )
-                    for (
-                        dataValue in d2.dataValueModule().dataValues()
-                            .byAttributeOptionComboUid().`in`(attributeOptionCombos)
-                            .byCategoryOptionComboUid().`in`(categoryOptionCombos)
-                            .byDataElementUid().eq(it.dataElement().uid()).blockingGet()
-                    ) {
-                        if (dataValue.state() != State.SYNCED) {
-                            state = State.TO_UPDATE
-                        }
-                    }
+                val possibleStates = repo.blockingGet().map {
+                    it.state()
                 }
 
-                d2.dataSetModule().dataSetCompleteRegistrations()
-                    .byDataSetUid().eq(dataSet.uid()).blockingGet().forEach {
-                    if (it.state() != State.SYNCED) {
-                        state = if (it.deleted() == true) {
-                            State.TO_UPDATE
-                        } else {
-                            it.state()
-                        }
-                    }
+                val state = if(possibleStates.contains(State.ERROR) ||
+                        possibleStates.contains(State.WARNING)){
+                    State.WARNING
+                }else if(possibleStates.contains(State.SENT_VIA_SMS) ||
+                        possibleStates.contains(State.SYNCED_VIA_SMS)){
+                    State.SENT_VIA_SMS
+                }else if(possibleStates.contains(State.TO_UPDATE) ||
+                        possibleStates.contains(State.TO_POST)){
+                    State.TO_UPDATE
+                }else{
+                    State.SYNCED
```

Comment:
very nice, u could use "when" kotlin if u want to make it more beautiful :D

Reply:
if we have time, we should pass the ktlint task.

Other than that, great!

Reply:
Last PR in the queue has the klint check :)